torch/_functorch/aot_autograd.py

```diff
@@ -3698,6 +3824,14 @@ def _compiled_autograd_key(ctx):
         @staticmethod
         def forward(ctx, *deduped_flat_tensor_args):
             args = deduped_flat_tensor_args
+
+            marked_dirty_inps = []
+            if fw_metadata.keep_input_mutations and fw_metadata.is_train:
```

Comment:
nit: `fw_metadata.is_train` is guaranteed to be true here, no need to check it (we're inside the `autograd.Function`)